OpenShift/helm/nexus-iq/templates/deployment.yaml

```diff
@@ -3,51 +3,55 @@ kind: Deployment
 metadata:
   name: {{ template "iqserver.fullname" . }}
   labels:
-    app: {{ template "iqserver.fullname" . }}
+    {{- include "iqserver.labels" . | nindent 4 }}
 spec:
-  replicas: 1
+  replicas: {{ .Values.replicaCount }}
   selector:
     matchLabels:
-      app: {{ template "iqserver.fullname" . }}
+      {{- include "iqserver.selectorLabels" . | nindent 6 }}
   template:
     metadata:
       labels:
-        app: {{ template "iqserver.fullname" . }}
+        {{- include "iqserver.selectorLabels" . | nindent 8 }}
     spec:
-      {{- if .Values.iq.imagePullSecret }}
+    {{- with .Values.imagePullSecrets }}
       imagePullSecrets:
-        - name: {{ template "iqserver.fullname" . }}-imagepull
-      {{- end }}
+        {{- toYaml . | nindent 8 }}
+    {{- end }}
+      serviceAccountName: {{ include "iqserver.serviceAccountName" . }}
+      securityContext:
+        {{- toYaml .Values.podSecurityContext | nindent 8 }}
       containers:
-      - name: {{ template "iqserver.fullname" . }}
-        image: {{ .Values.iq.imageName }}:{{ .Values.iq.imageTag }}
-        ports:
-        - containerPort: {{ .Values.iq.applicationPort }}
-        {{- if .Values.deployment.terminationGracePeriodSeconds }}
-        terminationGracePeriodSeconds: {{ .Values.deployment.terminationGracePeriodSeconds }}
-        {{- end }}
-        lifecycle:
-          {{- if .Values.deployment.postStart.command }}
-          postStart:
-            exec:
-              command: {{ .Values.deployment.postStart.command }}
```

Comment:
I see the postStart and preStart sections being removed here, but it still exists in the values. I'm concerned it may be missing now from this file that uses the values. I see other values preserved (volumeMounts, etc). How are pre-/post-Start applied?

Reply:
My mistake. I saw them preserved in the values.yaml for nxrm, not for iq. they've just disappeared, it seems. Was that intentional? is there another way to accomplish it?  We might want to closely review anything that appears to have been removed.

Reply:
This is anbother item I'd like to address in 'cleanup' issues. reconcile the templates and values along with removing any hard coding in a template